core/src/main/scala/com/metamx/tranquility/beam/Beam.scala

```diff
@@ -42,3 +61,27 @@ class DefunctBeamException(s: String, t: Throwable) extends Exception(s, t)
 {
   def this(s: String) = this(s, null)
 }
+
+object Beam
+{
+  /**
+    * Beam.index(xs) is like xs.zipWithIndex, but always returns an IndexedSeq so lookups by index will be efficient.
+    * @param xs elements
+    * @return indexed seq equivalent to xs.zipWithIndex
+    */
+  def index[A](xs: Seq[A]): IndexedSeq[(A, Int)] = {
```

Comment:
What happens if `Seq` has more than `Int.MaxValue` items? Both this and the BitSet would break